ocaml/xapi/storage_access.ml

```diff
@@ -39,6 +39,22 @@ let transform_storage_exn f =
       let backtrace = Backtrace.Interop.of_json "SM" backtrace in
       Backtrace.add e backtrace ;
       Backtrace.reraise e (Api_errors.Server_error (code, params))
+  | Storage_error (Sr_unhealthy (sr, health)) as e ->
+      let advice =
+        match health with
+        | Unavailable ->
+            "try reboot"
+        | Unreachable ->
+            "try again later"
+        | _health ->
+            ""
+      in
+      Backtrace.reraise e
+        (Api_errors.Server_error
+           ( Api_errors.sr_unhealthy
+           , [sr; Storage_interface.show_sr_health health; advice]
```

Comment:
What is the type of `sr`? The convention is to return API references in API errors.

Reply:
So I defined sr unhealthy as ` | Sr_unhealthy of string * sr_health` where the first string represents the uuid of the sr. The sr unhealthy error is generated like this
```
    let sr_uuid = Storage_interface.Sr.string_of sr in

          debug "%s: sr unhealthy because it is %s" __FUNCTION__
            (Storage_interface.show_sr_health health) ;
          Deferred.Result.fail
            Storage_interface.(Errors.Sr_unhealthy (sr_uuid, health))
 ```

Reply:
The `string_of` function there is just an identitiy function though, but if we chase all the way back to `xapi_sr` where the `scan` function is called, the input type of `sr` is indeed a `SR Ref.t`

Reply:
we can double check this by looking at the output of this error
```
The SR is currently unhealthy. See the suggestion on how to fix it.
sr: cc486fc6-4b25-59c1-6b02-14e7d371859e
health: Unavailable
fix: try reboot
```

Reply:
So it's a uuid, not a ref, otherwise it would have started with `OpaqueRef:` ;)

Reply:
That answered my month-long confusion about Ref and uuid....

Reply:
although I think in the end it will still be converted into uuid by the cli
```
xe sr-scan uuid=cc486fc6-4b25-59c1-6b02-14e7d371859e
The SR is currently unhealthy. See the suggestion on how to fix it.
sr: cc486fc6-4b25-59c1-6b02-14e7d371859e (GFS2 Storage Repository 1)
health: Unavailable
fix: try reboot
```

Reply:
..._if_ you use xe, yes.